src/test-utilities/TestProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
```

Comment:
Broke this apart from `test-utilities/react-testing` since it's not package specific/export issue

Reply:
I'm not sure what you mean by this? I'm not clear on the value of having this living in its own file

Reply:
Sorry I could have been more clear. This was the error
![Screen Shot 2019-07-12 at 11 09 55 AM](https://user-images.githubusercontent.com/24610840/61138522-ad888b00-a495-11e9-84a4-b0d95b7d0139.png)

![Screen Shot 2019-07-12 at 11 09 50 AM](https://user-images.githubusercontent.com/24610840/61138494-9e094200-a495-11e9-9a33-e866f1b4a346.png)

Eventually, I decided to move it into a separate file since `TestProvider` isn't test platform-specific. It can be used with `react-testing`, `enzyme`, `react-test-util`, etc.

Reply:
Following pairing: This TestProvider shall be the building block of our testing strategy. It is a generic component that can be used by both enzyme and react-testing's mounting systems.

legacy.tsx shall be updated to use this instead of providing its own version.

createPolarisContext shall be removed from our public API and instead 3rd party test configs (e.g. web) should use this TestProvider component as part of their test configuration, for both enzyme and react-testing setups


Related: Can we have a more specific name for this? `PolarisTestContextProvider` or something like that? It's not very clear what the identifier "TestProvider" would do if you're in web's test setup files.

Reply:
> Related: Can we have a more specific name for this? PolarisTestContextProvider or something like that? It's not very clear what the identifier "TestProvider" would do if you're in web's test setup files.

Yup that's definitely possible, it's actually similar to how I'm importing it right now 

```
import {..., TestProvider as PolarisTestProvider ,...} from '@shopify/polaris'
```

Reply:
I think there's some value in not naming our exports so generically people feel compelled to alias them :)